mama/c_cpp/src/c/mama.c

```diff
@@ -1208,11 +1218,33 @@ mama_closeCount (unsigned int* count)
     if (!--gImpl.myRefCount)
     {
 #ifdef WITH_ENTITLEMENTS
-        if( gEntitlementClient != 0 )
+        for (int bridgeCount=0 ; bridgeCount != gImpl.entitlements.count ; bridgeCount++)
         {
-            oeaClient_destroy( gEntitlementClient );
-            gEntitlementClient = 0;
+            mamaEntitlementLib* entLib = gImpl.entitlements.byIndex[bridgeCount];
+            if (entLib)
+            {
+                /* Remove from indexed list of entitlements libraries. */
+                gImpl.entitlements.byIndex[bridgeCount] = NULL;
+
+                /* Destroy the MAMA-level bridge and underlying implementation struct. */    
+                mamaEntitlementBridge_destroy(entLib->bridge);
+                entLib->bridge = NULL;
+
+                /* Destroy library struct. */
+                if (entLib->library)
+                {
+                    entLib->library = NULL;
```

Comment:
closseSharedLibrary() here (and the same for middleware and payload libs where necessary)